xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -2423,8 +2431,11 @@ public String getSkinFile(String filename, boolean forceSkinAction, XWikiContext
             }
 
```

Comment:
Aren't you missing the above 2 cases where a file can come from a skin document (in which case, you want the document version appended to the URL) or from a parent skin (i.e. baseskin, a filesystem folder -- in which case you want to append the same file size and/or last modification time parameters to the URL)?

Reply:
Yes only /resources/ use case seems to be supported right now but version support would make more sense in Resource#getURL implementations than here IMO.

Reply:
Now all cases are supported.